packages/@aws-cdk/aws-appmesh/lib/virtual-node-listener.ts

```diff
@@ -172,6 +191,7 @@ class VirtualNodeListenerImpl extends VirtualNodeListener {
         },
         healthCheck: this.healthCheck ? this.renderHealthCheck(this.healthCheck) : undefined,
         timeout: this.timeout ? this.renderTimeout(this.timeout) : undefined,
+        tls: this.tlsCertificate ? renderTls(scope, this.tlsCertificate) : undefined,
```

Comment:
This is a weird way to do that, passing the scope over to a helper method like that. What I would do instead is call `bind()` in the upper `bind()`, save the result to a local variable, and pass that to `renderTls` (which should handle the `undefined` case itself):

```ts
  public bind(scope: cdk.Construct): VirtualNodeListenerConfig {
    const tlsConfig = this.tlsCertificate?.bind(scope);
    return {
        tls: renderTls(tlsConfig),
        ...
```